cypress/e2e/34-template-credentials-setup.cy.ts

```diff
@@ -0,0 +1,81 @@
+import { CredentialsModal, MessageBox } from '../pages/modals';
+import { TemplateCredentialSetupPage } from '../pages/template-credential-setup';
+import { TemplateWorkflowPage } from '../pages/template-workflow';
+import { WorkflowPage } from '../pages/workflow';
+
+const templateWorkflowPage = new TemplateWorkflowPage();
+const templateCredentialsSetupPage = new TemplateCredentialSetupPage();
+const credentialsModal = new CredentialsModal();
+const messageBox = new MessageBox();
+const workflowPage = new WorkflowPage();
+
+const testTemplate = templateCredentialsSetupPage.testData.simpleTemplate;
+
+describe('Template credentials setup', () => {
+	it('can be opened from template workflow page', () => {
+		cy.intercept('GET', `https://api.n8n.io/api/templates/workflows/${testTemplate.id}`, {
+			fixture: testTemplate.fixture,
+		});
+
+		templateWorkflowPage.actions.visit(testTemplate.id);
+		templateCredentialsSetupPage.actions.enableFeatureFlag();
+		templateWorkflowPage.actions.clickUseThisWorkflowButton();
+
+		templateCredentialsSetupPage.getters
+			.title(`Setup 'Promote new Shopify products on Twitter and Telegram' template`)
+			.should('be.visible');
+	});
+
+	it('can be opened with a direct url', () => {
+		cy.intercept('GET', `https://api.n8n.io/api/templates/workflows/${testTemplate.id}`, {
+			fixture: testTemplate.fixture,
+		});
+		templateCredentialsSetupPage.actions.visit(testTemplate.id);
+
+		templateCredentialsSetupPage.getters
+			.title(`Setup 'Promote new Shopify products on Twitter and Telegram' template`)
+			.should('be.visible');
+	});
+
+	it('can be skip template creation', () => {
+		cy.intercept('GET', `https://api.n8n.io/api/templates/workflows/${testTemplate.id}`, {
+			fixture: testTemplate.fixture,
+		});
+		templateCredentialsSetupPage.actions.visit(testTemplate.id);
+
+		templateCredentialsSetupPage.getters.skipLink().click();
+		workflowPage.getters.canvasNodes().should('have.length', 3);
+	});
+
+	it('can create credentials and workflow from the template', () => {
```

Comment:
Think it would be nice to check if all the elements of setup page are there (all exepected credential pickers, notice has the right text, ...).